web/frontend/src/components/base.vue

```diff
@@ -79,17 +78,20 @@ const setDirection = (dir: Directions) => {
   direction.value = dir;
 };
 
-const stop = () => {
+const stop = async () => {
+  const bc = new BaseClient(props.client, props.name, { requestLogger: rcLogConditionally });
   stopped = true;
-  const req = new baseApi.StopRequest();
-  req.setName(props.name);
-  rcLogConditionally(req);
-  props.client.baseService.stop(req, new grpc.Metadata(), displayError);
+  try {
+    await bc.stop();
+  } catch (error) {
+    displayError(error as ServiceError);
+  }
 };
 
-const digestInput = () => {
+const digestInput = async () => {
   let linearValue = 0;
   let angularValue = 0;
+  const bc = new BaseClient(props.client, props.name, { requestLogger: rcLogConditionally });
```

Comment:
Is there a connectivity or some other reason why we're instantiating the BaseClient for every user interaction? It seems like a pretty substantial class so I'm somewhat concerned about putting the GC into overdrive / adding a few dozen ms or more to interaction responses on some lower power devices

Reply:
The BaseClient class is pretty light actually - most of the instantiation happens in the client that it receives as its first argument.

That said, there's no reason we couldn't just initiate it once. What's the right way to do this in Vue?

Reply:
Ah, good to know! It looks like it could be instantiated at the top of the script for this component and then available to the scope of all these functions.

Reply:
sounds good!